replica/table.cc

```diff
@@ -733,6 +735,31 @@ table::try_flush_memtable_to_sstable(lw_shared_ptr<memtable> old, sstable_write_
     co_return co_await with_scheduling_group(_config.memtable_scheduling_group, std::ref(try_flush));
 }
 
+future<> table::maybe_wait_for_sstable_count_reduction() {
+    while (true) {
+        if (_async_gate.is_closed()) {
+            co_return;
+        }
+        const auto sstable_runs_with_memtable_origin = boost::copy_range<std::unordered_set<utils::UUID>>(
+            *_main_sstables->all()
+            | boost::adaptors::filtered([] (const sstables::shared_sstable& sst) {
+                return sst->get_origin() == "memtable";
+            })
+            | boost::adaptors::transformed(std::mem_fn(&sstables::sstable::run_identifier)));
```

Comment:
Ah, so you're waiting for only flushed-but-not-compacted sstables, that's good.

Let's see if this filtering won't cause a stall. Assume 1 TB/shard, 160MB/sstable (all sstables, let's assume we're not in the bad situation). So we have 6000 sstables to filter through. It will cause a stall if each compare takes a microsecond. It should be much faster, so this code is safe.